src/cairoUtilFuncGen/serialisation.ts

```diff
@@ -13,19 +14,119 @@ export function serialiseReads(
   type: CairoType,
   readFelt: (offset: number) => string,
   readId: (offset: number) => string,
+  readAddress: (offset: number) => string,
+  uNread: (offset: number, nBits: number) => string,
 ): [reads: string[], pack: string] {
   const packExpression = producePackExpression(type);
   const reads: string[] = [];
   const packString: string = packExpression
     .map((elem: string | Read) => {
-      if (elem === Read.Felt) {
-        reads.push(readFelt(reads.length));
-        return `read${reads.length - 1}`;
-      } else if (elem === Read.Id) {
-        reads.push(readId(reads.length));
-        return `read${reads.length - 1}`;
-      } else {
-        return elem;
+      switch (elem) {
+        case Read.Felt:
+          reads.push(readFelt(reads.length));
+          return `read${reads.length - 1}`;
+        case Read.Id:
+          reads.push(readId(reads.length));
+          return `read${reads.length - 1}`;
+        case Read.Address:
+          reads.push(readAddress(reads.length));
+          return `read${reads.length - 1}`;
+        case Read.U8:
+          reads.push(uNread(reads.length, 8));
```

Comment:
I think all those Read.UN can be written as one if we put that `N` value as type.nBits. Somethin like 
```
    case Read.UN:
        reads.push(uNread(reads.length, type.nBits));
        return `read${reads.length - 1}`;
```

perhaps would be needed an `assert(type instance of CairoUint)` at case begin

Reply:
<img width="750" alt="Screenshot 2023-04-25 at 6 36 47 PM" src="https://user-images.githubusercontent.com/47271616/234285769-b6cef45e-7898-493e-84da-ac932c1f0f25.png">
Really? It doesn't seem to work

Reply:
Ye, you have to define it first in the enum

https://github.com/NethermindEth/warp/blob/6ee56532862c1e21d18fe1553731be0e3c1b9678/src/cairoUtilFuncGen/serialisation.ts#L136-L144

Reply:
got it